### PR TITLE
Fix fullscreen if there is no current webview (zim)

### PR DESCRIPTION
**Pull request by HanYoung-uwu**
Exit full screen if last tab closed or settings menu is bringing to the front. Disable full screen action if no tab opened or on settings menu.

#571

### CODE REVIEW

File: src/kiwixapp.h [hunk 1]
```diff
@@ -83,6 +83,7 @@ class KiwixApp : public QtSingleApplication
     QString getText(const QString &key) { return m_translation.getText(key); };
 
     bool isCurrentArticleBookmarked();
+    void setFullScreenDisable(bool disable) {mpa_actions[ToggleFullscreenAction]->setDisabled(disable);}
```
Comment (asashnov) [inline src/kiwixapp.h:86]: I don't think we need to introduce this new method here. It's better (more readable, even longer) already used approach:
```c
KiwixApp::instance()->getAction(KiwixApp::ToggleFullscreenAction)
    ->setDisabled(true);
```

**Review by asashnov — CHANGES_REQUESTED** [review]

**Review by asashnov — CHANGES_REQUESTED** [review]
Adding a new (shortcut) method will likely decrease the code readability:
https://github.com/kiwix/kiwix-desktop/pull/580#discussion_r563719152

**Comment by kelson42** [comment]
@HanYoung-uwu Are you ready for an other review?

File: src/tabbar.cpp [hunk 2]
```diff
@@ -192,6 +192,10 @@ void TabBar::closeTab(int index)
     removeTab(index);
     webview->close();
     delete webview;
+
+    // exit full screen on last tab closed
+    if(this->count() <= 2 &&KiwixApp::instance()->getAction(KiwixApp::ToggleFullscreenAction)->isChecked())
```
Comment (kelson42) [inline src/tabbar.cpp:197]: Please keep a space after the if keyword.

Comment (asashnov) [reply src/tabbar.cpp]: and after &&

Comment (kelson42) [reply src/tabbar.cpp]: done

**Comment by kelson42** [comment]
@asashnov Is this PR good to merge in your opinion?

**Comment by asashnov** [comment]
@HanYoung-uwu the code looks good, but I discovered strange (unwanted to me as a user) behaviour.

Steps to reproduce:
* run the program. Only library tab is shown;
* activate 'Main menu (...)' -> Settings

Actual result:
The settings tab is shown, but the program switched to fullscreen also.

Expected result:
The 'Settings' tab should be shown as usual, without the change of window size (current 'master' cb5cfdce is OK in that sense).

**Review by asashnov — CHANGES_REQUESTED** [review]
Code seems good, but Main menu -> Settings forces the app go fullscreen (see https://github.com/kiwix/kiwix-desktop/pull/580#issuecomment-770775880)

**Review by asashnov — CHANGES_REQUESTED** [review]
Commit message should contain Fixes #571  so GitHub can really associate this PR with the issue (the link in PR description works but is not making the association, it's just a mention of some issue to GitHub now).

And by the way, what we are going to achieve with this fix? As I understand, there was a crash, but it doesn't crash now:
https://github.com/kiwix/kiwix-desktop/issues/571#issuecomment-771032667

**Comment by HanYoung-uwu** [comment]
> Commit message should contain Fixes #571 so GitHub can really associate this PR with the issue (the link in PR description works but is not making the association, it's just a mention of some issue to GitHub now).
> 
> And by the way, what we are going to achieve with this fix? As I understand, there was a crash, but it doesn't crash now:
> [#571 (comment)](https://github.com/kiwix/kiwix-desktop/issues/571#issuecomment-771032667)

I'm also confused by this, IIRC, we pass nullptr to fullscreen. But as that doesn't crash the app, this MR doens't really fixed anything. I wonder if I close this MR instead.

**Comment by kelson42** [comment]
@HanYoung-uwu Why have you closed the PR?

**Comment by HanYoung-uwu** [comment]
> @HanYoung-uwu Why have you closed the PR?

Bad git practice. I always forgot to open a new branch for MR. Then I force sync the master with upstream.

**Comment by kelson42** [comment]
@HanYoung-uwu You should not open a new PR. You should modify your history in place in this current MR.

**Comment by HanYoung-uwu** [comment]
> @HanYoung-uwu You should not open a new PR. You should modify your history in place in this current MR.

I'm not open a new PR, I mean I forgot I have this PR try to merge my master into this master. And I force reset my master to this master to sync. In the process unintentionally closed the PR.

**Comment by kelson42** [comment]
@HanYoung-uwu First I would recommend you to not work on master. You should work in a branch. That would ease the whole operation. What the plan now?

**Comment by HanYoung-uwu** [comment]
> @HanYoung-uwu First I would recommend you to not work on master. You should work in a branch. That would ease the whole operation. What the plan now?

> I always forgot to open a new branch for MR

This PR needs more discussion I guess. I'm not very sure what is the expected result. The Outcome of my edits is if in settings or blank view kiwix-desktop will auto exit from fullscreen mode and disable it.

**Comment by kelson42** [comment]
@HanYoung-uwu I believe to have clarified the ticket. Look at Firefox to see how the top bar disappearance shoudl work.

**Comment by HanYoung-uwu** [comment]
So the current Full Screen function is implemented via `showFullScreen` in mainwindow class. I understand that @kelson42 wants to full screen the webview not the whole application window. I propose to connect the fullscreen action to tabbar instead. That way, we can full screen the webview, thus essentially making the web content full screen. Reviews appreciated.

**Pull request closed without merging**